src/main/java/com/google/sps/model/TasksResponse.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import java.util.List;
+
+/**
+ * Contains the summary tasks information that should be passed to the client, as well as the
+ * methods to generate these statistics.
+ */
+public final class TasksResponse {
+  private final List<String> taskListNames;
+  private final int tasksToComplete;
+  private final int tasksDueToday;
+  private final int tasksCompletedToday;
+  private final int tasksOverdue;
+
+  /**
+   * Create a TasksResponse instance
+   *
+   * @param taskListNames names of all task lists to display in a multiselect dropdown list on the
+   *     client.
+   * @param tasksToComplete number of tasks the user has yet to complete.
+   * @param tasksDueToday number of tasks the user has yet to complete for today.
+   * @param tasksCompletedToday number of tasks the user has marked as complete for today.
+   * @param tasksOverdue number of tasks that have due dates before today.
+   */
+  public TasksResponse(
+      List<String> taskListNames,
+      int tasksToComplete,
+      int tasksDueToday,
+      int tasksCompletedToday,
+      int tasksOverdue) {
```

Comment:
This constructor is just at the edge of being too complicated. If you were to add any more fields, I would recommend switching to a Builder pattern for initializing this object.

The concern here is that it would be very easy to get the order of the parameters wrong, and there is no way to see that at the call site. Unit tests should help catch bugs, but it's better when the code is self-documenting.

(You don't need to change it in this PR.)

Reply:
I agree, will definitely look into refactoring this using the Java annotation system you [linked](https://projectlombok.org/features/Builder) in our group chat.

Reply:
`RESOLVED` https://github.com/googleinterns/step7-2020/pull/105